### ISSUE TITLE
WARNING: at fs/inode.c:1084 discard_new_inode+0x6e/0x80

### ISSUE THREAD

**Issue by g2p** [koverstreet/bcachefs#690]
Kernel is bcachefs/master as of b37c1b07570da3a481dc913b44f53a06d259595e (rebased on slightly newer linus/master), and I get these warnings periodically from multithreaded programs like dua, fclones, nfsd.

Apparently discard_new_inode is called on an inode that isn't really new.  Caller was updated in d93ff5fa40b9db5f505d508336bc171f54db862e "bcachefs: Fix race path in bch2_inode_insert()".

```
void discard_new_inode(struct inode *inode)
{
    lockdep_annotate_inode_mutex_key(inode);
    spin_lock(&inode->i_lock);
    WARN_ON(!(inode->i_state & I_NEW));
    inode->i_state &= ~I_NEW;
    smp_mb();
    wake_up_bit(&inode->i_state, __I_NEW);
    spin_unlock(&inode->i_lock);
    iput(inode);
}
```

<details><summary>dmesg</summary>

```
------------[ cut here ]------------
WARNING: CPU: 1 PID: 26531 at fs/inode.c:1084 discard_new_inode+0x6e/0x80
Modules linked in: essiv(E) authenc(E) snd_seq_dummy(E) snd_hrtimer(E) xfs(E) dm_crypt(E) kyber_iosched(E) cmac(E) ccm(E) nls_utf8(E) wireguard(E) curve25519_x86_64(E) libcurve25519_generic(E) libchacha20poly1305(E) chacha_x86_64(E) poly1305_x86_64(E) ip6_udp_tunnel(E) udp_tunnel(E) ip6t_REJECT(E) nf_reject_ipv6(E) xt_hl(E) ip6t_rt(E) ipt_REJECT(E) nf_reject_ipv4(E) xt_multiport(E) xt_recent(E) nft_limit(E) xt_limit(E) xt_addrtype(E) xt_tcpudp(E) xt_conntrack(E) nf_conntrack(E) nf_defrag_ipv6(E) nf_defrag_ipv4(E) nft_compat(E) nf_tables(E) binfmt_misc(E) btrfs(E) blake2b_generic(E) nls_iso8859_1(E) iwlmvm(E) mac80211(E) snd_hda_codec_realtek(E) snd_hda_codec_generic(E) snd_hda_scodec_component(E) snd_hda_codec_hdmi(E) snd_hda_intel(E) snd_intel_dspcfg(E) libarc4(E) snd_hda_codec(E) snd_hwdep(E) snd_hda_core(E) nouveau(E) snd_pcm(E) iwlwifi(E) drm_ttm_helper(E) ttm(E) video(E) gpu_sched(E) drm_gpuvm(E) drm_exec(E) snd_seq(E) mxm_wmi(E) snd_seq_device(E) snd_timer(E) drm_display_helper(E) cfg80211(E)
 drm_kms_helper(E) snd(E) k10temp(E) intel_rapl_msr(E) soundcore(E) intel_rapl_common(E) wmi_bmof(E) kvm_amd(E) i2c_piix4(E) kvm(E) mac_hid(E) rapl(E) wmi(E) nfsd(E) drm(E) lockd(E) auth_rpcgss(E) grace(E) sunrpc(E) efi_pstore(E) nfnetlink(E) dmi_sysfs(E) ip_tables(E) x_tables(E) bcache(E) bridge(E) crct10dif_pclmul(E) stp(E) pata_acpi(E) crc32_pclmul(E) llc(E) polyval_clmulni(E) polyval_generic(E) nvme(E) ghash_clmulni_intel(E) sha512_ssse3(E) igb(E) sha256_ssse3(E) pata_jmicron(E) i2c_algo_bit(E) ahci(E) sha1_ssse3(E) dca(E) nvme_core(E) libahci(E) nvme_auth(E) ccp(E) xhci_pci(E) xhci_pci_renesas(E) dm_mirror(E) dm_region_hash(E) dm_log(E) msr(E) autofs4(E) aesni_intel(E) crypto_simd(E) cryptd(E)
CPU: 1 PID: 26531 Comm: dua-fs-walk-5 Tainted: G        W   E      6.10.0-rc3-g2p #84
Hardware name: To Be Filled By O.E.M. X570 Phantom Gaming 4/X570 Phantom Gaming 4, BIOS P5.61 02/22/2024
RIP: 0010:discard_new_inode+0x6e/0x80
Code: 8d bb 98 00 00 00 e8 e1 dc ca ff 4c 89 e7 e8 79 70 cf 00 48 89 df e8 41 fd ff ff 5b 41 5c 5d 31 c0 31 f6 31 ff e9 a2 7e cf 00 <0f> 0b eb bb 66 66 2e 0f 1f 84 00 00 00 00 00 0f 1f 00 f3 0f 1e fa
RSP: 0018:ffffaba4223935f0 EFLAGS: 00010246
RAX: 0000000000000000 RBX: ffff97b8973cce18 RCX: 0000000000000000
RDX: 0000000000000000 RSI: 0000000000000000 RDI: 0000000000000000
RBP: ffffaba422393600 R08: 0000000000000000 R09: 0000000000000000
R10: 0000000000000000 R11: 0000000000000000 R12: ffff97b8973ccea0
R13: ffff97b6ab017ab8 R14: ffffaba422393740 R15: ffff97b684548000
FS:  00007f7d895ff6c0(0000) GS:ffff97b9aec80000(0000) knlGS:0000000000000000
CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
CR2: 00007fbd02754000 CR3: 0000000154746000 CR4: 0000000000350ef0
Call Trace:
 <TASK>
 ? show_regs+0x70/0x90
 ? __warn+0x8c/0x150
 ? discard_new_inode+0x6e/0x80
 ? report_bug+0x186/0x1b0
 ? handle_bug+0x48/0x90
 ? exc_invalid_op+0x1c/0x80
 ? asm_exc_invalid_op+0x1f/0x30
 ? discard_new_inode+0x6e/0x80
 bch2_inode_insert+0x13c/0x150
 bch2_lookup_trans+0x5d6/0x5f0
 bch2_lookup+0xad/0x180
 ? srso_return_thunk+0x5/0x5f
 ? bch2_lookup+0xad/0x180
 __lookup_slow+0x85/0x130
 ? srso_return_thunk+0x5/0x5f
 walk_component+0x117/0x190
 path_lookupat+0x6a/0x1b0
 filename_lookup+0xde/0x1f0
 vfs_statx+0x95/0x1d0
 do_statx+0x68/0xc0
 __x64_sys_statx+0x6b/0xa0
 x64_sys_call+0x340/0x25e0
 do_syscall_64+0x68/0x120
 ? __fput+0x1b1/0x2e0
 ? srso_return_thunk+0x5/0x5f
 ? kmem_cache_free+0x3fc/0x420
 ? srso_return_thunk+0x5/0x5f
 ? __fput+0x1b1/0x2e0
 ? srso_return_thunk+0x5/0x5f
 ? __fput_sync+0x20/0x30
 ? srso_return_thunk+0x5/0x5f
 ? srso_return_thunk+0x5/0x5f
 ? syscall_exit_to_user_mode+0x7c/0x180
 ? srso_return_thunk+0x5/0x5f
 ? do_syscall_64+0x74/0x120
 ? srso_return_thunk+0x5/0x5f
 ? atime_needs_update+0x69/0x130
 ? srso_return_thunk+0x5/0x5f
 ? touch_atime+0x23/0x130
 ? srso_return_thunk+0x5/0x5f
 ? iterate_dir+0x1b6/0x230
 ? srso_return_thunk+0x5/0x5f
 ? __f_unlock_pos+0x16/0x20
 ? srso_return_thunk+0x5/0x5f
 ? __x64_sys_getdents64+0x112/0x140
 ? __x64_sys_getdents64+0x140/0x140
 ? srso_return_thunk+0x5/0x5f
 ? syscall_exit_to_user_mode+0x7c/0x180
 ? srso_return_thunk+0x5/0x5f
 ? do_syscall_64+0x74/0x120
 ? srso_return_thunk+0x5/0x5f
 ? syscall_exit_to_user_mode+0x7c/0x180
 ? srso_return_thunk+0x5/0x5f
 ? do_syscall_64+0x74/0x120
 ? do_syscall_64+0x74/0x120
 ? __x64_sys_statx+0x75/0xa0
 ? srso_return_thunk+0x5/0x5f
 ? syscall_exit_to_user_mode+0x7c/0x180
 ? srso_return_thunk+0x5/0x5f
 ? do_syscall_64+0x74/0x120
 ? do_syscall_64+0x74/0x120
 entry_SYSCALL_64_after_hwframe+0x55/0x5d
RIP: 0033:0x7f7d8971bf1e
Code: 6f 0e 00 ba ff ff ff ff 64 c7 00 16 00 00 00 e9 b9 fd ff ff e8 83 bf 01 00 0f 1f 00 f3 0f 1e fa 41 89 ca b8 4c 01 00 00 0f 05 <48> 3d 00 f0 ff ff 77 2a 89 c1 85 c0 74 0f 48 8b 05 c5 6e 0e 00 64
RSP: 002b:00007f7d895fc1f8 EFLAGS: 00000202 ORIG_RAX: 000000000000014c
RAX: ffffffffffffffda RBX: 00007f7d895fc320 RCX: 00007f7d8971bf1e
RDX: 0000000000000100 RSI: 00007f7d895fc460 RDI: 00000000ffffff9c
RBP: 0000000000000001 R08: 00007f7d895fc200 R09: 0000000000000007
R10: 0000000000000fff R11: 0000000000000202 R12: 00007f7d895fcb80
R13: 00007f7d402fc5c0 R14: 00000000000000a0 R15: 00007f7d895fc460
 </TASK>
---[ end trace 0000000000000000 ]---
```
</details>

**Comment by koverstreet** [comment]
Fixed by
1f06731fa910 bcachefs: Fix I_NEW warning in race path in bch2_inode_insert()
which is now in my bcachefs-for-upstream branch

**Comment by g2p** [comment]
This didn't stay fixed, I have boots with the warning that postdate v6.10-rc5 (PR was https://lore.kernel.org/linux-bcachefs/171907260669.30765.11576903904385555925.pr-tracker-bot@kernel.org/T/).  Though I think the issue has only shown up with bcachefs-testing; possibly it is triggered by "bcachefs: allocate inode by using alloc_inode_sb()".

**Comment by toddejohnson** [comment]
I think I've run into this again too(6.10.0-1-bcachefs-git-00140-gc84469634e82):

<details>
<summary> Trace</summary>

```
[   20.563368] ------------[ cut here ]------------
[   20.563372] WARNING: CPU: 2 PID: 768 at fs/inode.c:1084 discard_new_inode+0x5c/0x60
[   20.563385] Modules linked in: rpcrdma rdma_cm iw_cm ib_cm ib_core dimlib rfkill intel_rapl_msr intel_rapl_common intel_uncore_frequency intel_uncore_frequency_common x86_pkg_temp_thermal intel_powerclamp coretemp kvm_int
el kvm igb ipmi_ssif rndis_host iTCO_wdt cdc_ether vfat usbnet spi_nor crct10dif_pclmul crc32_pclmul ptp intel_pmc_bxt fat pps_core iTCO_vendor_support polyval_clmulni polyval_generic joydev mtd mii dca intel_lpss_pci gf128m
ul ghash_clmulni_intel sha512_ssse3 intel_lpss sha1_ssse3 i2c_i801 aesni_intel mousedev crypto_simd cryptd mei_me intel_pmc_core mei acpi_ipmi i2c_smbus video idma64 pcspkr rapl intel_vsec ipmi_si i2c_mux ipmi_devintf wmi ip
mi_msghandler intel_cstate acpi_cpufreq intel_uncore pmt_telemetry pmt_class mac_hid acpi_tad nfsd auth_rpcgss nfs_acl lockd grace sunrpc dm_mod loop nfnetlink ip_tables x_tables hid_generic usbhid uas usb_storage nvme nvme_
core spi_intel_pci ast sha256_ssse3 xhci_pci spi_intel i2c_algo_bit nvme_auth xhci_pci_renesas bcachefs libcrc32c crc32c_generic
[   20.563473]  crc32c_intel lz4_compress lz4hc_compress xor raid6_pq
[   20.563480] CPU: 2 PID: 768 Comm: nfsd Not tainted 6.10.0-1-bcachefs-git-00140-gc84469634e82 #1 a72a40d5967d40a0b3750397c448a76784f4441e
[   20.563485] Hardware name: Supermicro SYS-521R-T/X13SCL-IF, BIOS 1.3 03/01/2024
[   20.563487] RIP: 0010:discard_new_inode+0x5c/0x60
[   20.563494] Code: 00 f0 83 44 24 fc 00 be 03 00 00 00 48 8d bb 98 00 00 00 e8 c6 f3 cd ff 48 89 ef e8 6e ca a0 00 48 89 df 5b 5d e9 74 fd ff ff <0f> 0b eb c8 90 90 90 90 90 90 90 90 90 90 90 90 90 90 90 90 f3 0f
[   20.563496] RSP: 0000:ffffb63543723a48 EFLAGS: 00010246
[   20.563500] RAX: 0000000000000000 RBX: ffff9643a3e35a00 RCX: ffffb63543723a60
[   20.563502] RDX: 0000000000000001 RSI: 0000000000000001 RDI: ffff9643a3e35a88
[   20.563504] RBP: ffff9643a3e35a88 R08: ffffb63543723a60 R09: ffffffff9b649fe8
[   20.563506] R10: feb1b7e2716b2000 R11: 0000000000006840 R12: ffff9643aed7ad00
[   20.563508] R13: ffff96438995c000 R14: ffffb63543723ab0 R15: 0000000000000001
[   20.563510] FS:  0000000000000000(0000) GS:ffff9652bf700000(0000) knlGS:0000000000000000
[   20.563513] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[   20.563515] CR2: 00007f1d481cc970 CR3: 0000000ebb220000 CR4: 0000000000f50ef0
[   20.563517] PKRU: 55555554
[   20.563519] Call Trace:
[   20.563522]  <TASK>
[   20.563524]  ? discard_new_inode+0x5c/0x60
[   20.563529]  ? __warn.cold+0x8e/0xe8
[   20.563535]  ? discard_new_inode+0x5c/0x60
[   20.563544]  ? report_bug+0xff/0x140
[   20.563548]  ? handle_bug+0x3c/0x80
[   20.563554]  ? exc_invalid_op+0x17/0x70
[   20.563559]  ? asm_exc_invalid_op+0x1a/0x20
[   20.563565]  ? discard_new_inode+0x5c/0x60
[   20.563570]  ? discard_new_inode+0x1d/0x60
[   20.563575]  bch2_inode_insert+0x15d/0x170 [bcachefs ca577434c3460198342506a5b05523903e0ebff0]
[   20.563706]  bch2_vfs_inode_get+0x1c1/0x240 [bcachefs ca577434c3460198342506a5b05523903e0ebff0]
[   20.563831]  ? __pfx_nfsd_acceptable+0x10/0x10 [nfsd 19991b1b229a1ceae223ac39aa376dfd3631997c]
[   20.563912]  bch2_fh_to_dentry+0x43/0x80 [bcachefs ca577434c3460198342506a5b05523903e0ebff0]
[   20.564020]  exportfs_decode_fh_raw+0x7c/0x310
[   20.564029]  fh_verify+0x57f/0x740 [nfsd 19991b1b229a1ceae223ac39aa376dfd3631997c]
[   20.564102]  nfsd_statfs+0x2c/0x90 [nfsd 19991b1b229a1ceae223ac39aa376dfd3631997c]
[   20.564175]  nfsd3_proc_fsstat+0x39/0x70 [nfsd 19991b1b229a1ceae223ac39aa376dfd3631997c]
[   20.564252]  nfsd_dispatch+0xcf/0x220 [nfsd 19991b1b229a1ceae223ac39aa376dfd3631997c]
[   20.564321]  svc_process_common+0x4d2/0x6a0 [sunrpc e2840354d42d3e0952634ca91922cd49c8ffa96b]
[   20.564420]  ? __pfx_nfsd_dispatch+0x10/0x10 [nfsd 19991b1b229a1ceae223ac39aa376dfd3631997c]
[   20.564486]  svc_process+0x131/0x180 [sunrpc e2840354d42d3e0952634ca91922cd49c8ffa96b]
[   20.564571]  svc_recv+0x7f4/0x9b0 [sunrpc e2840354d42d3e0952634ca91922cd49c8ffa96b]
[   20.564652]  ? __pfx_nfsd+0x10/0x10 [nfsd 19991b1b229a1ceae223ac39aa376dfd3631997c]
[   20.564722]  nfsd+0x87/0xd0 [nfsd 19991b1b229a1ceae223ac39aa376dfd3631997c]
[   20.564788]  kthread+0xcf/0x100
[   20.564794]  ? __pfx_kthread+0x10/0x10
[   20.564798]  ret_from_fork+0x31/0x50
[   20.564805]  ? __pfx_kthread+0x10/0x10
[   20.564809]  ret_from_fork_asm+0x1a/0x30
[   20.564814]  </TASK>
[   20.564816] ---[ end trace 0000000000000000 ]---
[   21.635191] bcachefs (0fb7f58b-bec6-4201-ac25-949e55e3a59c): bch2_inode_peek(): error looking up inum 1:537082530: ENOENT_inode
[   21.635194] bcachefs (0fb7f58b-bec6-4201-ac25-949e55e3a59c): bch2_inode_peek(): error looking up inum 1:537082530: ENOENT_inode
[   21.636615] ------------[ cut here ]------------
[   21.636618] WARNING: CPU: 10 PID: 767 at fs/inode.c:1084 discard_new_inode+0x5c/0x60
[   21.636628] Modules linked in: rpcrdma rdma_cm iw_cm ib_cm
[   21.636631] bcachefs (0fb7f58b-bec6-4201-ac25-949e55e3a59c): bch2_inode_peek(): error looking up inum 1:537082530: ENOENT_inode
[   21.636633]  ib_core dimlib rfkill intel_rapl_msr intel_rapl_common intel_uncore_frequency intel_uncore_frequency_common x86_pkg_temp_thermal intel_powerclamp coretemp kvm_intel kvm igb ipmi_ssif rndis_host
[   21.636670] bcachefs (0fb7f58b-bec6-4201-ac25-949e55e3a59c): bch2_inode_peek(): error looking up inum 1:537082530: ENOENT_inode
[   21.636671]  iTCO_wdt
[   21.636673]  cdc_ether vfat usbnet spi_nor crct10dif_pclmul crc32_pclmul ptp intel_pmc_bxt fat pps_core iTCO_vendor_support polyval_clmulni polyval_generic joydev mtd mii dca intel_lpss_pci gf128mul ghash_clmulni_intel sh
a512_ssse3 intel_lpss sha1_ssse3 i2c_i801 aesni_intel mousedev crypto_simd cryptd mei_me intel_pmc_core mei acpi_ipmi i2c_smbus video idma64 pcspkr rapl intel_vsec ipmi_si i2c_mux ipmi_devintf wmi ipmi_msghandler intel_cstat
e acpi_cpufreq intel_uncore pmt_telemetry pmt_class mac_hid acpi_tad nfsd auth_rpcgss nfs_acl lockd grace sunrpc dm_mod loop nfnetlink ip_tables x_tables hid_generic usbhid uas usb_storage nvme nvme_core spi_intel_pci ast sh
a256_ssse3 xhci_pci spi_intel i2c_algo_bit nvme_auth xhci_pci_renesas bcachefs libcrc32c crc32c_generic crc32c_intel lz4_compress lz4hc_compress xor raid6_pq
[   21.636748] CPU: 10 PID: 767 Comm: nfsd Tainted: G        W          6.10.0-1-bcachefs-git-00140-gc84469634e82 #1 a72a40d5967d40a0b3750397c448a76784f4441e
[   21.636752] Hardware name: Supermicro SYS-521R-T/X13SCL-IF, BIOS 1.3 03/01/2024
[   21.636756] RIP: 0010:discard_new_inode+0x5c/0x60
[   21.636761] Code: 00 f0 83 44 24 fc 00 be 03 00 00 00 48 8d bb 98 00 00 00 e8 c6 f3 cd ff 48 89 ef e8 6e ca a0 00 48 89 df 5b 5d e9 74 fd ff ff <0f> 0b eb c8 90 90 90 90 90 90 90 90 90 90 90 90 90 90 90 90 f3 0f
[   21.636765] RSP: 0018:ffffb6354371ba68 EFLAGS: 00010246
[   21.636768] RAX: 0000000000000000 RBX: ffff9643aeff8f00 RCX: ffffb6354371ba80
[   21.636770] RDX: 0000000000000001 RSI: 0000000000000001 RDI: ffff9643aeff8f88
[   21.636772] RBP: ffff9643aeff8f88 R08: ffffb6354371ba80 R09: ffffffff9b649fe8
[   21.636773] R10: f988b856886aa100 R11: 0000000000006840 R12: ffff9643880d1e00
[   21.636774] R13: ffff9643b2780000 R14: ffffb6354371bad0 R15: 0000000000000001
[   21.636776] FS:  0000000000000000(0000) GS:ffff9652bfb00000(0000) knlGS:0000000000000000
[   21.636778] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[   21.636780] CR2: 00007f1d28011548 CR3: 0000000ebb220000 CR4: 0000000000f50ef0
[   21.636782] PKRU: 55555554
[   21.636783] Call Trace:
[   21.636792]  <TASK>
[   21.636793]  ? discard_new_inode+0x5c/0x60
[   21.636798]  ? __warn.cold+0x8e/0xe8
[   21.636802]  ? discard_new_inode+0x5c/0x60
[   21.636811]  ? report_bug+0xff/0x140
[   21.636814]  ? handle_bug+0x3c/0x80
[   21.636819]  ? exc_invalid_op+0x17/0x70
[   21.636823]  ? asm_exc_invalid_op+0x1a/0x20
[   21.636827]  ? discard_new_inode+0x5c/0x60
[   21.636831]  ? discard_new_inode+0x1d/0x60
[   21.636881] bcachefs (0fb7f58b-bec6-4201-ac25-949e55e3a59c): bch2_inode_peek(): error looking up inum 1:537082530: ENOENT_inode
[   21.636836]  bch2_inode_insert+0x15d/0x170 [bcachefs ca577434c3460198342506a5b05523903e0ebff0]
[   21.636989]  bch2_vfs_inode_get+0x1c1/0x240 [bcachefs ca577434c3460198342506a5b05523903e0ebff0]
[   21.637080]  ? __pfx_nfsd_acceptable+0x10/0x10 [nfsd 19991b1b229a1ceae223ac39aa376dfd3631997c]
[   21.637173]  bch2_fh_to_dentry+0x43/0x80 [bcachefs ca577434c3460198342506a5b05523903e0ebff0]
[   21.637257]  exportfs_decode_fh_raw+0x7c/0x310
[   21.637264]  fh_verify+0x57f/0x740 [nfsd 19991b1b229a1ceae223ac39aa376dfd3631997c]
[   21.637321]  nfsd3_proc_getattr+0x6e/0x110 [nfsd 19991b1b229a1ceae223ac39aa376dfd3631997c]
[   21.637382]  nfsd_dispatch+0xcf/0x220 [nfsd 19991b1b229a1ceae223ac39aa376dfd3631997c]
[   21.637440]  svc_process_common+0x4d2/0x6a0 [sunrpc e2840354d42d3e0952634ca91922cd49c8ffa96b]
[   21.637548]  ? __pfx_nfsd_dispatch+0x10/0x10 [nfsd 19991b1b229a1ceae223ac39aa376dfd3631997c]
[   21.637600]  svc_process+0x131/0x180 [sunrpc e2840354d42d3e0952634ca91922cd49c8ffa96b]
[   21.637666]  svc_recv+0x7f4/0x9b0 [sunrpc e2840354d42d3e0952634ca91922cd49c8ffa96b]
[   21.637730]  ? __pfx_nfsd+0x10/0x10 [nfsd 19991b1b229a1ceae223ac39aa376dfd3631997c]
[   21.637784]  nfsd+0x87/0xd0 [nfsd 19991b1b229a1ceae223ac39aa376dfd3631997c]
[   21.637837]  kthread+0xcf/0x100
[   21.637841]  ? __pfx_kthread+0x10/0x10
[   21.637845]  ret_from_fork+0x31/0x50
[   21.637850]  ? __pfx_kthread+0x10/0x10
[   21.637853]  ret_from_fork_asm+0x1a/0x30
[   21.637857]  </TASK>
[   21.637858] ---[ end trace 0000000000000000 ]---
```
</details>

I wonder if the same patch could be applied and help.  Another kernel package...

**Comment by g2p** [comment]
@koverstreet Please reopen, syzbot is correct

**Comment by g2p** [comment]
(now fixed in 99c87fe0f584f8d778a323141504d1ba5c89a4a5 "bcachefs: fix incorrect i_state usage")
